### DISCUSSION TITLE
Store Pending message

### DISCUSSION THREAD

**Discussion by Vezna14** [dotnet/MQTTnet#1992] [Q&A]
Hello, 

I have a use case where I need to persist messages until i have the confirmation that a client have received it. 
From my understanding, the Quality of Service options can do that but the messages a queued in the RAM. 
I would like to know if there is a way to change that so the messages will be stored in a file or an internal message queue like MSMQ. It will probably be less efficient but in my use case, keeping messages in memory is not suitable.

**Comment by ChrisProlls** [comment]
Same for me :)
I would like to backed the queing with a storage (SQL Server, CosmosDb whatever) so that if the service restart it doesn't lost his messages. Isn't that possible ?
